.github/workflows/dependabot-pr.yml

```diff
@@ -0,0 +1,48 @@
+name: Dependabot PR
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  dependabot:
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+    if: github.triggering_actor == 'dependabot[bot]'
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          token: ${{ secrets.PAT }}
+
+      - name: update code
+        run: |
+          input="${{ github.event.pull_request.title }}"
+
+          regex="[Bb]ump (.*) from (.*) to (.*)"
+          if [[ ! "$input" =~ $regex ]]; then
+            echo "unexpected pattern"
+            exit 1
+          fi
+
+          crate_name="${BASH_REMATCH[1]}"
+          old_version="${BASH_REMATCH[2]}"
+          new_version="${BASH_REMATCH[3]}"
+          echo "crate_name: $crate_name, old_version: $old_version, new_version: $new_version"
+
+          ./scripts/cargo-for-all-lock-files.sh -- update -p $crate_name:$old_version --precise $new_version
+          ./scripts/cargo-for-all-lock-files.sh -- tree
```

Comment:
nit: i don't think `tree` is needed here. the preceding `update` should have updated any `Cargo.lock`s...

Reply:
actually we will need to do both strict rule and relaxed rule. the first is for multiple version of crates in our repo. the second is for crates that are not in the `programs/sbf/Cargo.toml`. 

https://github.com/anza-xyz/agave/pull/1496/commits/2898cc094c030bf60dcc73b2721e735e5d2ddec2